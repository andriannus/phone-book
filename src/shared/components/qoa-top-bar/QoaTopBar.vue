<template>
  <div class="TopBar">
    <div class="TopBar-logo">
      <span class="FontSize-large">
        <strong>Qoala Test</strong>
      </span>
    </div>

    <div class="TopBar-actions">
      <div class="TopBar-action">
        <button class="Button Button--primary" type="button">
          Color
        </button>
      </div>

      <div class="TopBar-action">
        <button class="Button Button--primary Button--outlined" type="button">
          Cities
        </button>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: "QoaTopBar",
};
</script>
