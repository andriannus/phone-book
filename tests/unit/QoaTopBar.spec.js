import { shallowMount } from "@vue/test-utils";

import QoaTopBar from "@/shared/components/qoa-top-bar/QoaTopBar.vue";

describe("QoaTopBar.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(QoaTopBar);
  });

  it("should create", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should emit when color button clicked", () => {
    const sortByColorButton = wrapper.find("#BtnSortByColor");

    sortByColorButton.trigger("click");

    expect(wrapper.emitted("sorted")).toBeTruthy();
  });

  it("should emit with `color` argument when color button clicked", () => {
    const sortByColorButton = wrapper.find("#BtnSortByColor");

    sortByColorButton.trigger("click");

    expect(wrapper.emitted("sorted")[0]).toEqual(["color"]);
  });

  it("should emit when cities button clicked", () => {
    const sortByCitiesButton = wrapper.find("#BtnSortByCities");

    sortByCitiesButton.trigger("click");

    expect(wrapper.emitted("sorted")).toBeTruthy();
  });

  it("should emit with `cities` argument when color button clicked", () => {
    const sortByCitiesButton = wrapper.find("#BtnSortByCities");

    sortByCitiesButton.trigger("click");

    expect(wrapper.emitted("sorted")[0]).toEqual(["cities"]);
  });
});
