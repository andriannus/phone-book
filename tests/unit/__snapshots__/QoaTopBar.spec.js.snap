// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QoaTopBar.vue should created 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {},
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="TopBar"
      >
        <div
          class="TopBar-logo"
        >
          <span
            class="FontSize-large"
          >
            <strong>
              Qoala Test
            </strong>
          </span>
        </div>
        <div
          class="TopBar-actions"
        >
          <div
            class="TopBar-action"
          >
            <button
              class="Button Button--primary Button--outlined"
              id="BtnSortByColor"
              type="button"
            >
               Color 
            </button>
          </div>
          <div
            class="TopBar-action"
          >
            <button
              class="Button Button--primary Button--outlined"
              id="BtnSortByCities"
              type="button"
            >
               Cities 
            </button>
          </div>
        </div>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {},
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.2",
  },
  "__functionalEmits": Object {},
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
    "sortUser": [Function],
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
